transcript/src/main/java/com/transcriptanalyzer/transcript/Tree.java

```diff
@@ -0,0 +1,93 @@
+package com.transcriptanalyzer.transcript;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+// Assume we are producing one tree for a certain transcript in the ArrayList of transcripts
+public class Tree {
+    private Node treeRootNode;
+    private HashMap<String, Integer> counts;
+
+    // class for tree representing intents as it's nodes
+    // intents organized in order of their turn in their respective transcript
+    public Tree(ArrayList<ArrayList<ArrayList<String>>> transcripts) {
+        // root of tree
+        this.treeRootNode = new Node("Root", null);
+        Node start = this.treeRootNode;
+        this.counts = new HashMap<String, Integer>();
+
+        // go through each transcript, then each turn (representing each message intent pair)
+        // and add it as a subIntent node of the treeNode, then update treeNode down
+        for (ArrayList<ArrayList<String>> transcript : transcripts) {
```

Comment:
Consider whether creating the tree and updating the count of the intents at each node are the same responsibility. If not they should be different methods to follow the single responsibility principle.

Reply:
If by that you mean a seprate method with an algorithm that does the cumulative counting of occurrences, I agree.